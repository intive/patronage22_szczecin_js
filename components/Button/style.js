import styled, { css } from 'styled-components'
import Icon from '../Icon/Icon'

export const StyledButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0px 4px 4px rgba(81, 81, 81, 0.16);
  border: none;
  border-radius: 8px;
  height: 2.5rem;
  padding: 0.5rem 1.5rem;
  background: #515151;
  cursor: pointer;
  color: #FFFFFF;

  &:hover {
    background: #4a4949;
  }

  &:active {
    background: #3a3939;
  }

  &:disabled {
    opacity: 0.3;
    background: #515151;
    cursor: default;
    color: #FFFFFF;
  }

  ${props => props.text && css`
    background: transparent;
    color: #515151;
    box-shadow: none;

    &:hover {
      background: transparent;
      color: #000000;
    }

    &:active {
      background: transparent;
      color: #3a3939;
    }

    &:disabled {
      opacity: 0.3;
      background: #bababa9e;
      cursor: default;
      color: #515151;
    }
  `}

  ${props => props.outline && css`
    border: 1px solid #515151;
    &:hover {
      border: 1px solid #000000;
    }
  `}
`

export const StyledIcon = styled(Icon)`
  margin-right: 0.25rem;
`
