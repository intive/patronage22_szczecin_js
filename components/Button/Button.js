import { StyledButton, StyledIcon } from './style'

const Button = (props) => {
  return (
    <StyledButton
      onClick={props.onClick}
      disabled={props.disabled}
      {...props}
    >
      {props.icon
        ? (
          <>
            <StyledIcon name={props.icon} />
            {props.children}
          </>
          )
        : props.children}
    </StyledButton>
  )
}

export default Button
