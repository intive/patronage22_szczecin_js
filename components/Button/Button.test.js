/* globals describe, expect, it */

import { render } from '@testing-library/react'
import Button from './Button'

describe('Button', () => {
  it('renders contained button with icon', () => {
    const button = render(<Button icon='crop_original'>New board</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toBeEnabled()
  })

  it('renders contained button without icon', () => {
    const button = render(<Button>New board</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toBeEnabled()
  })

  it('renders button with icon', () => {
    const button = render(<Button text icon='crop_original'>Continue</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toBeEnabled()
  })

  it('renders button without icon', () => {
    const button = render(<Button text>Continue</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toBeEnabled()
  })

  it('renders contained disabled button with icon', () => {
    const button = render(<Button icon='crop_original' disabled>New board</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toHaveAttribute('disabled')
    expect(button.getByRole('button')).toBeDisabled()
  })

  it('renders contained disabled button without icon', () => {
    const button = render(<Button disabled>New board</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toHaveAttribute('disabled')
    expect(button.getByRole('button')).toBeDisabled()
  })

  it('renders disabled button with icon', () => {
    const button = render(<Button text icon='crop_original' disabled>Continue</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toHaveAttribute('disabled')
    expect(button.getByRole('button')).toBeDisabled()
  })

  it('renders disabled button without icon', () => {
    const button = render(<Button text disabled>Continue</Button>)
    expect(button.container).toMatchSnapshot()
    expect(button.getByRole('button')).toHaveAttribute('disabled')
    expect(button.getByRole('button')).toBeDisabled()
  })
})
