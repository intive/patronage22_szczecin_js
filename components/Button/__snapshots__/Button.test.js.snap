// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button with icon 1`] = `
<div>
  <button
    class="sc-bdvvtL dzUoKp"
  >
    <span
      class="material-icons sc-gsDKAQ jUyrhl"
    >
      crop_original
    </span>
    Continue
  </button>
</div>
`;

exports[`Button renders button without icon 1`] = `
<div>
  <button
    class="sc-bdvvtL dzUoKp"
  >
    Continue
  </button>
</div>
`;

exports[`Button renders contained button with icon 1`] = `
<div>
  <button
    class="sc-bdvvtL jESiNr"
  >
    <span
      class="material-icons sc-gsDKAQ jUyrhl"
    >
      crop_original
    </span>
    New board
  </button>
</div>
`;

exports[`Button renders contained button without icon 1`] = `
<div>
  <button
    class="sc-bdvvtL jESiNr"
  >
    New board
  </button>
</div>
`;

exports[`Button renders contained disabled button with icon 1`] = `
<div>
  <button
    class="sc-bdvvtL jESiNr"
    disabled=""
  >
    <span
      class="material-icons sc-gsDKAQ jUyrhl"
    >
      crop_original
    </span>
    New board
  </button>
</div>
`;

exports[`Button renders contained disabled button without icon 1`] = `
<div>
  <button
    class="sc-bdvvtL jESiNr"
    disabled=""
  >
    New board
  </button>
</div>
`;

exports[`Button renders disabled button with icon 1`] = `
<div>
  <button
    class="sc-bdvvtL dzUoKp"
    disabled=""
  >
    <span
      class="material-icons sc-gsDKAQ jUyrhl"
    >
      crop_original
    </span>
    Continue
  </button>
</div>
`;

exports[`Button renders disabled button without icon 1`] = `
<div>
  <button
    class="sc-bdvvtL dzUoKp"
    disabled=""
  >
    Continue
  </button>
</div>
`;
